aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py

```diff
@@ -614,13 +704,23 @@ async def credential_match_schema(credential: VCRecord, schema_id: str) -> bool:
     for cred_schema_id in credential.schema_ids:
         if cred_schema_id == schema_id:
             return True
-    for cred_type in credential.types:
+    expanded = jsonld.expand(json.loads(credential.cred_value))
+    types = JsonLdProcessor.get_values(
+        expanded[0],
+        "@type",
+    )
+    for cred_type in types:
         if cred_type == schema_id:
             return True
     return False
 
 
-async def apply_requirements(req: Requirement, credentials: Sequence[VCRecord]) -> dict:
+async def apply_requirements(
+    req: Requirement,
+    credentials: Sequence[VCRecord],
+    profile: Profile,
```

Comment:
I think we should avoid passing the profile everywhere and try to only pass the required instances